dhall/src/semantics/resolve/resolve.rs

```diff
@@ -303,6 +322,51 @@ pub(crate) fn download_http_text(_url: Url) -> Result<String, Error> {
     panic!("Remote imports are not supported on wasm yet")
 }
 
+#[cfg(all(not(target_arch = "wasm32"), feature = "reqwest"))]
+pub(crate) fn cors_check(
+    origin: Option<String>,
+    remote: &Url,
+) -> Result<(), Error> {
```

Comment:
I'd prefer that `cors_check` take origin as a String, and that testing whether `origin` is `None` be done outside of it.